crates/rmf_site_editor/src/mapf_rse/negotiation/mod.rs

```diff
@@ -0,0 +1,360 @@
+    negotiation_params: Res<NegotiationParams>,
+    mut negotiation_debug_data: ResMut<NegotiationDebugData>,
+    mut compute_negotiation_task: Query<(Entity, &mut NegotiationTask)>,
+    current_level: Res<CurrentLevel>,
+    grids: Query<(Entity, &Grid)>,
+    child_of: Query<&ChildOf>,
+    robots: Query<(Entity, &NameInSite, &Pose, &Affiliation<Entity>), With<Robot>>,
+    robot_descriptions: Query<(&DifferentialDrive, &CircleCollision)>,
+    tasks: Query<(&RobotTask, &GoToPlace)>,
+) {
+    if negotiation_request.len() == 0 {
+        return;
+    }
+    if !compute_negotiation_task.is_empty() {
+        let Ok((entity, negotiation_task)) = compute_negotiation_task.single_mut() else {
+            error!("Error: Found multiple negotiation tasks!");
+            return;
+        };
+
+        if let NegotiationTaskStatus::InProgress { .. } = negotiation_task.status {
+            warn!("Negotiation requested while another negotiation is in progress");
+            return;
+        }
+        // Computation is done, we can despawn the previous negotiation task
+        commands.entity(entity).despawn();
+    }
+    negotiation_debug_data.selected_negotiation_node = None;
+
+    // Occupancy
+    let mut occupancy = HashMap::<i64, Vec<i64>>::new();
+    let mut cell_size = 1.0;
+    let grid = grids
+        .iter()
+        .filter_map(|(grid_entity, grid)| {
+            if let Some(level_entity) = current_level.0 {
+                if child_of
+                    .get(grid_entity)
+                    .is_ok_and(|co| co.parent() == level_entity)
+                {
+                    Some(grid)
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .next();
+    match grid {
+        Some(grid) => {
+            cell_size = grid.cell_size;
+            for cell in grid.occupied.iter() {
+                occupancy.entry(cell.y).or_default().push(cell.x);
+            }
+            for (_, column) in &mut occupancy {
+                column.sort_unstable();
+            }
+        }
+        None => {
+            occupancy.entry(0).or_default().push(0);
+            warn!("No occupancy grid found, defaulting to empty");
+        }
+    }
+
+    // Agent
+    let mut agents = BTreeMap::<String, Agent>::new();
+    // Only loop tasks that have specified a valid robot
+    for (task, go_to_place) in tasks.iter() {
+        // Identify robot
+        let robot_name = task.robot();
+        for (robot_entity, robot_site_name, robot_pose, robot_group) in robots.iter() {
+            if robot_name == robot_site_name.0 {
+                // Match location to entity
+                for (_, location_name, Point(anchor_entity)) in locations.iter() {
+                    if location_name.0 == go_to_place.location {
+                        let Ok((_, _, goal_transform)) = anchors.get(*anchor_entity) else {
```

Comment:
If you are not going to use a variable (i.e. here doing `_`) you should just remove it from the `Query` altogether